spec/views/user_index_spec.rb

```diff
@@ -0,0 +1,38 @@
+require 'rails_helper'
+
+RSpec.feature 'User Index Page', type: :feature do
+  before do
+    @user1 = FactoryBot.create(:user, name: 'User1', photo: 'https://placehold.co/400')
+    @user2 = FactoryBot.create(:user, name: 'User2', photo: 'https://placehold.co/400')
+    FactoryBot.create(:post, author: @user1)
+    FactoryBot.create(:post, author: @user2)
+  end
+
+  scenario 'I can see the username of all other users' do
+    visit users_path
+
+    expect(page).to have_content('User1')
+    expect(page).to have_content('User2')
+  end
+
+  scenario 'I can see the profile picture for each user' do
+    visit users_path
+
+    expect(page).to have_css("img[src*='https://placehold.co/400']")
+  end
+
+  scenario 'I can see the number of posts each user has written' do
+    visit users_path
+
+    expect(page).to have_content('Number of posts: 1', count: 2)
+  end
+
```

Comment:
- [ ] Some of the tests are failing, Please make sure that all tests should pass without any errors.

![image](https://github.com/faisalakb/R_blog_app/assets/83803414/9d7259f1-892f-4d75-804b-879ce5179d50)